apis/python/tests/test_dataframe.py

```diff
@@ -848,6 +849,9 @@ def test_write_categorical_types(tmp_path):
                 "B": pd.Categorical(
                     ["a", "b", "a", "b"], ordered=False, categories=["b", "a"]
                 ),
+                "C": pd.Categorical(
+                    ["a", "b", "a", "b"], ordered=False, categories=["a", "b"]
+                ),
```

Comment:
this needs to also test categoricals that contain primitives other than string. Commonly see int and bool in real-world data.